drivers/pwm/pwm_stm32.c

```diff
@@ -1,11 +1,10 @@
 /*
  * Copyright (c) 2016 Linaro Limited.
+ * Copyright (c) 2020 Teslabs Engineering S.L.
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#define DT_DRV_COMPAT st_stm32_pwm
```

Comment:
Added to the top. The reason for doing that is to prevent accidental direct usage of DT macros, as it usually makes drivers not capable of handling an arbitrary number of instances.

Reply:
As mentioned, I'm totally open to change this but it would be better to discuss and update it widely then.